### DISCUSSION TITLE
Player bodies on abrupt disconnect are disappearing

### DISCUSSION THREAD

**Discussion by Kiskadee-dev** [Vdauphin/HeartsAndMinds#1417] [Question and answer] [answered]
Hi again, today we did the first test with the tickets on our server and we've noticed that when a dead player disconnects, their dead bodies will vanish in some cases and we can't add more tickets to the disconnected player.

It's not consistent, it happens mostly when their game crashes, other players didn't had this problem disconnecting normally and found their bodies later.

Of course the players could bring enemies to add more tickets when they were still alive, but what should we do when they doesn't have any tickets left and we can't retrieve their bodies due to arma shenanigans?

I've observed this behaviour twice in the latest version, but we never used tickets before so I can't say it's a bug or just a misconfiguration on our part. I've seen it happen when a player died and his game crashed, also with another one that got kicked by battleye, their bodies disappeared from the map and we have no idea on how to bring them back without wiping the save.

Any ideas on how to give them tickets or how to stop this behaviour?

**Comment by Vdauphin** [comment, marked as answer]
Hello,

> It's not consistent, it happens mostly when their game crashes,

I am bit surprise that your game crash that often. It is not an intended behavior so may be you should remove the mod creating that issue

> Of course the players could bring enemies to add more tickets when they were still alive, but what should we do when they doesn't have any tickets left and we can't retrieve their bodies due to arma shenanigans?

See http://vdauphin.github.io/HeartsAndMinds/InGame-documentation#get-new-tickets it is working even when they don't have tickets, they just need to be connected

>  their bodies disappeared from the map

This behavior is from Arma/ACE which remove people when they disconnect even in respawn screen

> Any ideas on how to give them tickets 

https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/respawn/addTicket.sqf

> how to stop this behaviour?

I will test a fix which basically copy the dead body after the game delete it. I hope it will work

Cheers

**Reply by Vdauphin** [reply]
See https://github.com/Vdauphin/HeartsAndMinds/pull/1418
